cloud/vsphere/machineactuator.go

```diff
@@ -729,14 +741,17 @@ func getKubeadmToken() (string, error) {
 // the appropriate reason/message on the Machine.Status. If not, such as during
 // cluster installation, it will operate as a no-op. It also returns the
 // original error for convenience, so callers can do "return handleMachineError(...)".
-func (vc *VsphereClient) handleMachineError(machine *clusterv1.Machine, err *apierrors.MachineError) error {
+func (vc *VsphereClient) handleMachineError(machine *clusterv1.Machine, err *apierrors.MachineError, eventAction string) error {
 	if vc.machineClient != nil {
 		reason := err.Reason
 		message := err.Message
 		machine.Status.ErrorReason = &reason
 		machine.Status.ErrorMessage = &message
 		vc.machineClient.UpdateStatus(machine)
 	}
+	if eventAction != "" {
+		vc.eventRecorder.Eventf(machine, corev1.EventTypeWarning, "Failed"+eventAction, "%v", err.Reason)
```

Comment:
I think it should be `"Failed "+eventAction` so that you get events like "Failed Create" instead of "FailedCreate".

Reply:
In that case should it be more specific, like `FailedCreateMachine`?

Reply:
In my opinion, it should not contain the word "Machine" because the `kind` field on the event is already Machine. Including it in the reason is redundant. When this event fires, you'll see this (for example):
   ```
   Kind: Machine
   Name: <Machine name>.<Machine uid>
   Type: Warning (there are only Normal and Warning types)
   Source: vsphere-machine-controller
   Reason: FailedCreate
   Message: InvalidConfiguration
   ```

You can't quite tell that this is what's going to happen by reading this code, but the event recorder uses the machine object here to get the `kind` and `name` fields for the event.

Reply:
thanks. this lgtm.